scripts/jenkins/cloud/ansible/roles/deploy_caasp_terraform/tasks/deploy_caasp.yml

```diff
@@ -0,0 +1,341 @@
+    ssh -o StrictHostKeyChecking=no sles@{{item}} sudo update-ca-certificates
+  loop: "{{ worker_ips.value.values() | list }}"
+
+- name: Management Machine | Update /etc/hosts on masters to add entry to the identity public domain name
+  shell: |
+    set -e
+    ssh -o StrictHostKeyChecking=no sles@{{item}} "echo {{resolved_public_vip }} {{ public_endpoint_domain }} | sudo tee -a /etc/hosts"
+  loop: "{{ master_ips.value.values() | list }}"
+  when: (resolved_public_vip is defined) and (resolved_public_vip|length > 0)
+
+- name: Management Machine | Update /etc/hosts on workers to add entry to the identity public domain name
+  shell: |
+    set -e
+    ssh -o StrictHostKeyChecking=no sles@{{item}} "echo "{{resolved_public_vip }} {{ public_endpoint_domain }}" | sudo tee -a /etc/hosts"
+  loop: "{{ worker_ips.value.values() | list }}"
+  when: (resolved_public_vip is defined) and (resolved_public_vip|length > 0)
+
+- name: Management Machine | Run skuba init
+  shell: |
+    set -e
+    cd {{ caasp_openstack_path }}
+    {{ skuba_init_command }} | tee -a skuba_init.log
+  register:
+    skuba_init_command_results
+  failed_when: " not ('[init] configuration files written to' in skuba_init_command_results.stdout)"
+
+- name: Management Machine | Move openstack.conf
+  shell: |
+    set -e
+    cd {{ caasp_openstack_path }}
+    {{ skuba_mv_openstack_conf_command }}
+  register:
+    skuba_mv_openstack_conf_command_results
+
+- name: Management Machine | Run skuba bootstrap
+  shell: |
+    set -e
+    eval `ssh-agent -s`
+    ssh-add
+    cd {{ caasp_openstack_path }}/{{ stack_name }}-cluster
+    {{ item }} | tee -a skuba_bootstrap.log
+  loop: "{{ skuba_boostrap_commands }}"
+  register:
+    skuba_bootstrap_commands_results
+  failed_when: " not ('[bootstrap] successfully bootstrapped core add-ons' in skuba_bootstrap_commands_results.stdout)"
+
+- name: Set expected total nodes in the cluster
+  set_fact:
+    total: "{{ num_masters | int + num_workers | int }}"
+
+- name: Set expected total nodes joining the cluster after bootstrap
+  set_fact:
+    total_join: "{{ num_masters | int + num_workers | int - 1 }}"
+
+- name: Management Machine | Run skuba join
+  shell: |
+    set -e
+    eval `ssh-agent -s`
+    ssh-add
+    cd {{ caasp_openstack_path }}/{{ stack_name }}-cluster
+    {{ item }} | tee -a skuba_join.log
+  loop: "{{ skuba_join_commands }}"
+  register:
+    skuba_join_commands_results
+
+- name: Management Machine | track the count of the nodes that join the cluster successfully
+  set_fact:
+    count: "{{ count | default(0) | int + 1 }}"
+  when: "'node successfully joined the cluster' in item"
+  loop: "{{ skuba_join_commands_results.results|map(attribute='stdout')|list }}"
+
+- name: Management Machine | Fail when the expected number of nodes do not join the cluster
+  fail:
+    msg: "Not all expected nodes have successfully joined the cluster"
+  when: "count | int != total_join | int"
+
```

Comment:
I did a deployment with num_masters=2 and num_workers=2.
https://ci.suse.de/blue/organizations/jenkins/openstack-crowbar/detail/openstack-crowbar/923/pipeline/158
It failed at this point. For some reason skuba printed `node successfully joined the cluster` twice. But `skuba cluster status` looks alright.

```
sles@caasp-management-20200417095828:~/caasp/deployment/openstack/auto-caasp-20200417095828-cluster> cat skuba_join.log 
[join] applying states to new node
[join] applying states to new node
[join] node successfully joined the cluster
[join] applying states to new node
[join] node successfully joined the cluster
sles@caasp-management-20200417095828:~/caasp/deployment/openstack/auto-caasp-20200417095828-cluster> skuba cluster status
NAME                                       STATUS   ROLE     OS-IMAGE                              KERNEL-VERSION           KUBELET-VERSION   CONTAINER-RUNTIME   HAS-UPDATES   HAS-DISRUPTIVE-UPDATES   CAASP-RELEASE-VERSION
caasp-master-auto-caasp-20200417095828-0   Ready    master   SUSE Linux Enterprise Server 15 SP1   4.12.14-197.37-default   v1.16.2           cri-o://1.16.1      <none>        <none>                   <none>
caasp-master-auto-caasp-20200417095828-1   Ready    master   SUSE Linux Enterprise Server 15 SP1   4.12.14-197.37-default   v1.16.2           cri-o://1.16.1      no            no                       4.1.2
caasp-worker-auto-caasp-20200417095828-0   Ready    <none>   SUSE Linux Enterprise Server 15 SP1   4.12.14-197.37-default   v1.16.2           cri-o://1.16.1      no            no                       4.1.2
caasp-worker-auto-caasp-20200417095828-1   Ready    <none>   SUSE Linux Enterprise Server 15 SP1   4.12.14-197.37-default   v1.16.2           cri-o://1.16.1      no            no                       4.1.2

```

Reply:
It's also strange that skuba did not set the role for the workers. But `--role worker` was in the command line. But that is a skuba problem.

Reply:
I'm not able to reproduce the issue:

I have tried using num_masters=3 in this example run: 
  https://ci.suse.de/job/openstack-caaspv4/5/console
I have tried using num_masters=2 in this example run: 
  https://ci.suse.de/job/openstack-caaspv4/6/console

Both the test runs passed.

Not sure why it has failed in your case. I'm building a crowbar environment in ECP to see if I can reproduce this issue there.

Yeah about the worker role.. I did notice that and it's a skuba issue.

Reply:
Maybe this task is not totally reliable. The next task counts the "Ready" nodes from `skuba cluster status` - maybe that's enough?

Reply:
Updated this task to just print the message as we check the status in the next task.